plugins/catalog-graph/src/components/EntityRelationsGraph/useEntityRelationGraph.ts

```diff
@@ -98,19 +100,21 @@ export function useEntityRelationGraph({
     entities,
     rootEntityRefs,
     maxDepth,
-    relations,
+    includeRelation,
     kinds,
     entityFilter,
     requestEntities,
   ]);
 
-  const filteredEntities = useMemo(
-    () =>
-      entityFilter
-        ? pickBy(entities, (value, _key) => entityFilter(value))
-        : entities,
-    [entities, entityFilter],
-  );
+  const filteredEntities = useMemo(() => {
+    if (loading) {
+      return {};
+    }
+
+    return entityFilter
+      ? pickBy(entities, (value, _key) => entityFilter(value))
+      : entities;
+  }, [loading, entities, entityFilter]);
```

Comment:
not sure if it could make sense to ignore the loading state, so that we avoid some flickering if the data is reloaded. Feel free to ignore if this introduces side effects 

```suggestion
  const filteredEntities = useMemo(() => {
    return entityFilter
      ? pickBy(entities, (value, _key) => entityFilter(value))
      : entities;
  }, [entities, entityFilter]);
```

Reply:
Ironically, it'll flicker less now.
The upstream `entities` will contain a bunch of entities. When the user changes the query, a new set of entities will be fetched (and `loading` will be true), but the old ones will remain in the `entities`, so leaving the UI with a progress spinner on top flickers less than "emptying" it with this `if (loading)` ☺️